pyccel/codegen/printing/fcode.py

```diff
@@ -1383,14 +1383,18 @@ def _print_FunctionDef(self, expr):
 
         for i,arg in enumerate(expr.arguments):
             if expr.arguments_inout[i]:
-                dec = Declare(arg.dtype, arg, intent='inout')
-
+                if isinstance(arg, FunctionAddress):
+                    dec = FuncAddressDeclare(arg, intent='inout')
```

Comment:
Is this possible?

Reply:
maybe in c if you like to change to which function the pointer is pointing to and then  return it, here i don't think it is. I'am gonna remove it

Reply:
But here we are translating from fortran. If you do:
```
def f(a):
    a  = new_func

b = old_func
f(b)
```
Then b is not modified

Reply:
what i mean is for:
```
def f(a):
    a = new_func
    return a
```
we should have the ability to change a inside the function
this is kind of irrelevant to your question

Reply:
If you mean:
```
def f(a):
    a = new_func
    return a
```
Then for fortran this would be translated by something like:
```
def f(a):
    a = new_func
    Out_0001 = a
    return Out_0001
```
so a is still an `in` not an `inout`. If you want to be able to do this, we may need to pass by value. I don't know if this is possible for procedures?
@yguclu  do you know what the standard says about this?

Reply:
I don't think that the `intent` keyword makes any sense for argument functions in Fortran...

Reply:
I've just checked, specifying the `intent` attribute for a dummy procedure is not allowed. It cannot be passed by `value` either.